site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,1305 @@
+# SQL to Substrait tutorial
+
+To provide an introduction to Substrait for those already familiar with SQL,
+this tutorial walks through the basics of Substrait by translating basic SQL
+queries. We'll present the Substrait in JSON form so it is (somewhat) readable
+for newcomers.
+
+!!! note
+    Substrait is currently only defined with Protobuf. The JSON provided here is the
+    Protobuf JSON output, but it is *not* the official Substrait text format. Eventually,
+    Substrait will define it's own human-readable text format, but for now this
+    tutorial will make due with what Protobuf provides.
+
+Substrait is designed to communicate plans (mostly logical plans). Those plans
+contain types, schemas, expressions, extensions, and relations. We'll look at 
+them in that order, going from simplest to most complex until we can construct
+full plans.
+
+![](substrait_components.svg)
```

Comment:
One minor nit about this image.  It shows four categories: types, relations, expressions, and functions.

I don't normally think of "functions" on the same level as the other three. Although, I'm also not sure exactly what "functions" is referring to, and maybe this is ok.

An expression can have a function invocation.  For example, a `ScalarFunction` message or a `WindowFunction` message.  But these are just a type of expression and so don't really deserve their own box.

On the other hand, a plan has a list of functions in the extension set (e.g. defines the URL/name/anchor).  If you're referring to this I suppose it's ok but in my mental picture I think of things like "read types" and "advanced extensions" at the same level of significance (e.g. not quite as significant as types/relations/expressions).

Anyways, I'm not sure anything here needs to change, it just seemed slightly off to me and I wanted to mention it.  If you're happy with it then we can stick with it as is.

Reply:
Yeah, this diagram is adapted from one of Jacques presentations. It's not really meant to accurately describe the relationships between each entity (except for the fact that plans are the top-level thing). It just gives an idea of what kinds of things are in plans.

I think it's fine for now, but if someone wants to design a better alternative, I have no qualms with replacing this.